src/m365/commands/login.ts

```diff
@@ -199,32 +203,36 @@ class LoginCommand extends Command {
         auth.connection.cloudType = CloudType.Public;
       }
 
-      try {
-        await auth.ensureAccessToken(auth.defaultResource, logger, this.debug);
-        auth.connection.active = true;
-      }
-      catch (error: any) {
-        if (this.debug) {
-          await logger.logToStderr('Error:');
-          await logger.logToStderr(error);
-          await logger.logToStderr('');
-        }
-
-        throw new CommandError(error.message);
-      }
+      await this.obtainAccessToken(logger);
```

Comment:
Since there's a chance that we're going to start logging in, why are we refreshing the access token here already? We should only do it when we determined that the current connection is ok.

Reply:
This is part of the login function, which is only called when a new login functionality needs to be started. The logic to check whether a new login functionality should be started whether only the accessToken should be renewed is done in the commandAction itself.

Please let me know if you have another vision to this so we can align